content/posts/Arnab_Das_GSoC_phase1.md

```diff
@@ -0,0 +1,178 @@
+
+![Simulator API flow](/images/Arnab_GSoC23/simulator_flow.png)
+
+#### - fixing bugs in the simulator
+
+- while implementing the api endpoints in the simulator I came across multiple bugs related to the tabs, circuit data generation for save, switching of circuit, deletion of circuit, creation of new circuits etc. All these issues occurred due to the development of the simulator without loading data from the api endpoints.
+- previously DOM manipulation was used to handle the loading, switching, deletion & creation of circuits in the scopes but with the implementation in vueJs, I moved all the tab related data to pinia store and manipulated the data in store to perform the switching, deletion, creation of circuits. This drastically reduced the usage of DOM manipulation and made the simulator more stable.
+- Similarly the bugs regarding circuit creation and deletion were fixed by creation of new independent components for those functions and using the store to manipulate the data.
+
+### V. Implementation of the new features in the vue simulator
+
+#### - replacing all confirm() alert() prompt() with custom vue components
+
+Previously the simulator used the default browser confirm(), alert() & prompt() for the confirmation of deletion of circuits, renaming of circuits, saving of circuits etc. This was replaced with custom vue components to make the simulator more user-friendly.
+
+![Alert](/images/Arnab_GSoC23/alert.png)
+![Confirm](/images/Arnab_GSoC23/confirm.png)
+![Prompt](/images/Arnab_GSoC23/prompt.png)
+
+#### - update project details from simulator after creation of new project
+
+Previously when a new project was created the user was redirected to the project details edit page to update the project details. But to make the simulator independent of the main repository, it was necessary to allow users to update project details from the simulator itself which also made the user experience better.
+
+this was achieved with a dialog box to update the project details from the simulator.
+
+##### - creation of wysiwyg editor for project description
+
+at the time of implementing the update project details dialog box, there was no pre-setup wysiwyg editor available which was compatible with our project. So I created a custom wysiwyg editor using the [tiptap](https://www.tiptap.dev/) library as a vue component.
+
+![Update Project Details](/images/Arnab_GSoC23/update_project_details.png)
+
+#### - Implementation of Import and Export of circuits as `.cv` files
+
+With the introduction of the import and export of circuits as .cv files, users will be able to import and export their circuits as .cv files. This will allow users to share their circuits with other users and also allow users to import circuits from other users. Also, this would help the users to keep a local backup of their circuits.
```

Comment:
```suggestion
With the introduction of the import and export of circuits as `.cv` files, users will be able to import and export their circuits as `.cv` files. This will allow users to share their circuits with other users and also allow users to import circuits from other users. Also, this would help the users to keep a local backup of their circuits.
```